enhancements/builds/remove-jenkins-playload.md

```diff
@@ -0,0 +1,195 @@
+Release notes will also need to inform admins that Jenkins images will need to be mirrored separately from the OpenShift payload.
+
+### Risks and Mitigations
+
+**Risk**: Jenkins imagestreams fail to import when the cluster is upgraded.
+
+*Mitigation*: Imagestreams don't report new references in status until the import is complete.
+Failures to import the new Jenkins imagestream will eventually fire an alert, unless the cluster admin skips the imagestream.
+
+**Risk**: Jenkins imagestreams are not updated in disconnected clusters.
+
+*Mitigation*: Documentation will need to provide new mirroring instructions for admins planning to upgrade.
+Release notes should also clearly indicate that Jenkins is no longer provided in the payload and needs to be mirrored separately.
+
+## Design Details
+
+### Open Questions [optional]
+
+None.
+
+### Test Plan
+
+This proposal is in the provisional state - test plan is not necessary at this time.
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Tech Preview -> GA
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Removing a deprecated feature
+
+The Jenkins pipeline strategy has been officially deprecated since OCP 4.1.
+This proposal does not permanently disable the strategy - rather it changes how we deliver an on-top application.
+Documentation will be updated to reflect areas where customers are impacted (mainly for network restricted environments).
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the Jenkins imagestream will be updated to reflect the new non-payload image by the Samples operator.
+
+On downgrade, the Jenkins imagestream will be restored to its prior payload image.
+
+### Version Skew Strategy
+
+Imagestream imports should not be impacted by version skew - imagestreams delivered by the Samples
+operator are not updated until the new version of the Samples operator is deployed and wins leader election.
+
+## Implementation History
+
+2021-07-20: Provisional enhancement draft
+
+## Drawbacks
+
+Removing Jenkins from the payload will help with the mainenance burden, because Jenkins fixes will not need to be backported to each OCP release.
```

Comment:
per my comments above, you may still need to maintain an image per OCP version, but at least you'll have more flexibility in doing the backports and shipping it, outside the OCP cadence.

Reply:
Spoke with @akram on this topic - we are already comfortable adding backwards compatibility to the Jenkins plugins. For instance, some of oauth plugin bits have code that supports 4.2 and earlier versions. Our current practice is to keep the same plugin versions across all supported releases.

I think this proposal will need to identify a versioning scheme so that if we do need to break backwards compatibility, we can do so safely.

Reply:
@bparees doesn't every operator have this problem? Have we established a pattern for OLM operator teams to maintain branches/versions for backwards compatiblity? Or are we mapping new territory here?

Reply:
first, we need to distinguish between talking about the operand image and the operator image.

With respect to operator images:  teams version their operators, always.  And they can decide which OCP catalog a given operator version goes into ("v1.1 of my operator only goes into the v4.7 OCP catalog"), though in theory an OCP 4.8 cluster can consume the v4.7 OCP catalog, but normally it does not.

With respect to operands:  The operator contains image references to the images it needs for operands. Those are normally SHA pinned when the operator bundle is constructed, so the operator is tied to a specific operand image.  However the operator can also be written to consume its operands in a non-SHA pinned way, or even in a way where you choose which operand image/version you want when you instantiate whatever thing the operator is providing.

but tldr: yes, operators and their associated operand are normally versioned.

As for how operator teams handle this..... yes they would normally maintain release branches.  That is what the logging operator team does, for example, but of course they are coming from the OCP world.  Some of the less OCP-coupled operator teams maintain branches named based on their operand version, though.

Reply:
Proposal has been updated to include branching and versioning scheme. We will carry backport branches when breaking changes are introduced.